### DISCUSSION TITLE
V2.15 RC1 |  Overdue changes | PD improvement

### DISCUSSION THREAD

**Discussion by Ralim** [Ralim/IronOS#943] [General]
This is a really good release.
Releasing as RC1 as would like to call for translation updates from the community.
Will elevate to full release once translations have settled down.

As always, thank you for using this firmware 😄 

## Feature upgrades:

* Multi-lingual firmware combinations now exist for Pinecil
* More fine grained voltage controlled options
* USB-PD improvements
* More configuration options for power pulse
* All font / character encoding has been very reworked
* More translation updates than one can count 
* More languages 😱 

### Kudos

I have most likely missed people here, but serious thankyou to all of the contributers to this firmware; and to all of the testers and users who raise issues.

@playday3008 
@t-8ch 
@HanaO00 
@xyzz 
@devydd 
@federicodilo 
@alvinhochun 
@morusn 
@TechnoLoft
@KITT3000 
@skishida 
@alvarosamudio 

<hr /><em>This discussion was created from the release <a href='https://github.com/Ralim/IronOS/releases/tag/v2.15-rc1'>V2.15 RC1 |  Overdue changes | PD improvement</a>.</em>

**Comment by elschopi** [comment]
Unfortunately, for me this release is making compatibility with USB PD worse. Powerbank that with the previous stable relase (from January) negotiated 12V 30W now only negotiates 9V 18W. 
Anker wall plug supply still doesn't negotiate correctly.
All in all, I think I'll stick with the January version.
Though, thanks for the effort :)

**Reply by Ralim** [reply]
Hello,

Could you please post the specifications of your power bank that is not negotiating 12V, as well as what iron you are using?

Some power adapters are still causing grief, especially for the TS80P. I'm yet to have a reproducible case to diagnose, so would also love any specs you can share of units that dont work.

**Reply by elschopi** [reply]
USB PD wall plug: Anker PowerPort Speed PD30, modell A2014
USB PD powerbank, 30W: PowerAdd Energy Cell II 26800

Hardware is a TS80P, PCB Version: (2)

**Reply by elschopi** [reply]
Might as well be some esoteric timing issues 🙈  
It's not the biggest problem in the world :P 
Unfortunately I don't have a PD analyzer at hand, else I'd be able to provide more information. 
Is there maybe a "debug mode" for the iron, where information can be displayed on screen?

**Comment by pedrolamas** [comment]
I just installed this on my TS80P, working fine with the OnePlus 9 official charger and the CHOETECH PD 61W USB C Charger! 🙂

**Comment by vini-nu** [comment]
Noticed degradation with TS80P PCB4 and it's official box-included EU PD2.0+QC3 45W (model QC09) charger. It is stuck on live-lock Miniware logo blinking and is unusable. 

TS80P powered by Lenovo loptop Type-C (PD?) 20V3.25A;15V3A;9V2A;5V2A charger goes into loop cycle with 3 iterations. Every iteration Miniware logo and my custom artwork is shown, after that 5V is negotiated and fails to go into 9V. Before update 9V was used, but charger had to be plugged into wall after connecting TS80P.

Used this RC version and tried manually compiled version from current master ( https://github.com/Ralim/IronOS/commit/1a0b542ae6fd1094306ac26f10119ed96641d5f0 ), last time I used 2.14.1 version.

Other power sources like 5V loptop USB2 and type-c smartphone works fine.

EDIT: It is now fixed with https://github.com/Ralim/IronOS/issues/949#issuecomment-830496035 build.

---

TS100 PCB2 works with no problems. Nice works!

**Comment by HanaO00** [comment]
Very interesting the multi language mode for the Pinecil, the TS80, TS80P and TS100 will have this possibility in the next updates?

**Reply by Ralim** [reply]
The smaller flash sizes for those units limits what options are easy to add for the multilingual firmware. At the moment it's not being built for them due to the extra size consumed.

**Comment by XaocuHKa** [comment]
Hey Ralim! Appreciate your work!!!

I flashed my first TS100 with 2.10 back in 2020 and still enjoying it.
Got another TS100, flashed with 2.15 and it doesnt go to 450C. Tried on friends TS100, and its the same thing. Heats up around 410 and doesnt go higher. Tried different supply 19V and 24V, no difference. Flashing back to 2.10 helps to resolve the thing.

Not sure if thats an issue, but maybe my feedback might be useful.

**Reply by HanaO00** [reply]
I just did a test with a Pinecil, boost temperature set to 450°C and I have the same issue.

**Reply by Ralim** [reply]
Ah yes, this changed when we improved the accuracy of the temp sensing processing.
The Iron has a maximum temperature above the internal temperature it can measure.
A while ago (around 2.13/2.14) I took a large sample set of tips and used measurements from them + the Hakko patent to build a nicer lookup table for uV <-> temperature for the tip.
While this table goes all the way up to 500C, the op-amp+ADC in the iron caps out around 420C or so.
The maximum temperature you are able to regulate to is the max measureable range by the op-amp + the internal handle temperature (cold junction temperature).

This re-measurement was done to bring the settled temperature far far closer to reality and improve the accuracy.

If you take your old iron and braised a thermocouple to the end of the tip, and then measured it on both firmwares you will find it caps out at around the same max temperature (+- a few C).

**Comment by Trezona** [comment]
Working great on my TS100.
Thanks very much for this.

**Comment by DarioB92** [comment]
Hello, I bought a 20V to usb-c adapter ( are used to convert laptop's powerbricks to usbc) , the device is capable to negotiate up to 12/15 V 3A or 20v 2.25A (tested), but the ts80p only takes 9V. There are resons for this? Thanks

**Reply by Ralim** [reply]
This firmware (2.15) is rather old.
Is the TS80P using PD or QC (Its in Debug menu on newer firmwares).
If its falling back to QC that would be the problem.
If its using PD then it will need to see 12V @ 2.7A or higher for it to select 12V otherwise it will use 9V@2A.
If the PD module does not advertise a 12V fixed PDO, but has PPS it can use that as well. If it does not have a 12V Fixed PDO and no PPS then it naturally cant select 12V.

**Reply by DarioB92** [reply]
I updated the firmware, I'm in PD.
The adapter uses a IP6518 ic, in datasheet I see in features and description 12V@3A , in Type-C port and USB PD protocol: "USB PD support output of : 5V/2.4A, 9V/2.4A, 12V/2.3A,15V/2A, 20V2A"
Probably it's due to chip's low current.
Thanks
